pkg/controllers/member/serviceexport/controller.go

```diff
@@ -256,6 +264,65 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 	return ctrl.Result{}, nil
 }
 
+func (r *Reconciler) setAzureRelatedInformation(ctx context.Context, service *corev1.Service, export *fleetnetv1alpha1.InternalServiceExport) error {
+	export.Spec.Type = service.Spec.Type
+	if service.Spec.Type != corev1.ServiceTypeLoadBalancer {
+		return nil
+	}
+	export.Spec.IsInternalLoadBalancer = service.Annotations[objectmeta.AzureLoadBalancerInternalAnnotation] == "true"
+	if export.Spec.IsInternalLoadBalancer {
+		// no need to populate the PublicIPResourceID and IsDNSLabelConfigured which are only applicable for external load balancer
+		return nil
+	}
+
+	serviceKObj := klog.KObj(service)
+	if len(service.Status.LoadBalancer.Ingress) == 0 {
+		// Assuming once the service status is updated, the controller will be triggered again.
+		klog.V(2).InfoS("The load balancer IP is not assigned yet", "service", serviceKObj)
+		return nil
+	}
+
+	if service.Status.LoadBalancer.Ingress[0].IP == "" {
+		err := errors.New("the service ingress is not nil but with empty IP")
+		klog.ErrorS(controller.NewUnexpectedBehaviorError(err), "Failed to get the load balancer IP from service", "service", serviceKObj, "status", service.Status)
+		return nil
+	}
+
+	pip, err := r.lookupPublicIPResourceIDByLoadBalancerIP(ctx, service)
+	if err != nil {
+		return err
+	}
+	if pip == nil {
+		klog.V(2).InfoS("The public IP is in the progressing", "service", serviceKObj, "ip", service.Status.LoadBalancer.Ingress[0].IP)
+		// Assuming once the service status is updated, the controller will be triggered again in instead of retrying here
+		// to avoid sending Azure requests.
+		return nil
+	}
+	export.Spec.PublicIPResourceID = pip.ID
+	// Note the user can set the dns label via the Azure portal or Azure CLI without updating service.
+	// This information may be stale as we don't monitor the public IP address resource.
+	export.Spec.IsDNSLabelConfigured = pip.Properties != nil && pip.Properties.DNSSettings != nil && pip.Properties.DNSSettings.DomainNameLabel != nil
+	return nil
+}
+
+// TODO: can improve the performance by caching the public IP address resource ID.
+func (r *Reconciler) lookupPublicIPResourceIDByLoadBalancerIP(ctx context.Context, service *corev1.Service) (*armnetwork.PublicIPAddress, error) {
+	serviceKObj := klog.KObj(service)
+	pips, err := r.AzurePublicIPAddressClient.List(ctx, r.ResourceGroupName)
```

Comment:
cx may also set pip rg in the service annotation: https://github.com/kubernetes-sigs/cloud-provider-azure/blob/master/pkg/provider/azure_loadbalancer.go#L3492